api/types/types.proto

```diff
@@ -2276,3 +2293,24 @@ message NetworkRestrictionsV4 {
     NetworkRestrictionsSpecV4 Spec = 5
         [ (gogoproto.nullable) = false, (gogoproto.jsontag) = "spec" ];
 }
+
+// RecoveryCodes holds a user's recovery code information.
+message RecoveryCodes {
+    // Kind describes the kind of resource.
+    string Kind = 1;
+    // Version is resource version.
+    string Version = 2;
+    // Codes hold a list of recovery codes.
+    repeated RecoveryCode Codes = 3 [ (gogoproto.nullable) = false ];
+    // Created is when the recovery codes were generated.
+    google.protobuf.Timestamp Created = 4
+        [ (gogoproto.stdtime) = true, (gogoproto.nullable) = false ];
+}
```

Comment:
I think `RecoveryCodes` should be converted into a proper resource since it is being stored as one. Additionally, I would update the key prefix from `web/users/[userID]/recoverycodes` to `web/recoverycodes/[userID]`, and make `RecoveryCodes.Metatdata.Name = userID`.
```suggestion
// RecoveryCodes holds a user's recovery code information.
message RecoveryCodesV1 {
    // Kind is the resource kind.
    string Kind = 1 [ (gogoproto.jsontag) = "kind" ];
    // SubKind is an optional resource subkind.
    string SubKind = 2 [ (gogoproto.jsontag) = "sub_kind,omitempty" ];
    // Version is the resource version.
    string Version = 3 [ (gogoproto.jsontag) = "version" ];
    // Metadata is the resource metadata.
    Metadata Metadata = 4 [ (gogoproto.nullable) = false, (gogoproto.jsontag) = "metadata" ];
    // Spec is the resource spec.
    RecoveryCodesSpecV1 Spec = 5 [ (gogoproto.nullable) = false, (gogoproto.jsontag) = "spec" ];
}

// RecoveryCodesSpecV1 is the recovery codes spec.
message RecoveryCodesSpecV1 {
    // Codes hold a list of recovery codes.
    repeated RecoveryCode Codes = 3 [ (gogoproto.nullable) = false ];
    // Created is when the recovery codes were generated.
    google.protobuf.Timestamp Created = 4
        [ (gogoproto.stdtime) = true, (gogoproto.nullable) = false ];
}
```

Reply:
converted to proper resource 

>  Additionally, I would update the key prefix from web/users/[userID]/recoverycodes to web/recoverycodes/[userID]

I kept the key format as is b/c other user resource follow this format and it is required to follow this format when we [DeleteUser](https://github.com/gravitational/teleport/blob/master/lib/services/local/users.go#L325), it deletes all user related resource by `web/users/<userID>/*`

`web/users/<userID>/params`
`web/users/<userID>/mfa/<mfaID>`
`web/users/<userID>/pwd`
`web/users/<userID>/recoverycodes`